portfolio/src/main/webapp/script-map.js

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+let map;
+const flagIcon = 'https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png';
+
+/* Editable marker that displays when a user clicks in the map. */
+let editMarker;
+
+/** Creates a map that shows travel markers and allows user to input markers. */
+function createMap() {
+  map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 39.329858, lng: -76.620540}, zoom: 5});
+  
+  // When the user clicks in the map, show a marker with a text box the user can edit.
+  map.addListener('click', (event) => {
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+
+  createMarkerForDisplay(
+      39.329858, -76.620540,
+      'I go to school here.', 'Johns Hopkins University', false)
+  createMarkerForDisplay(
+      32.003407, 118.734956,
+      'This is where I was born.', 'Nanjing', false)
+  createMarkerForDisplay(
+      37.423829, -122.092154,
+      'This is where my internship was supposed to be, but now it\'s remote.', 'Google', false);
+  createMarkerForDisplay(
+      23.117487, -82.373375,
+      'I went there with my friends. It was great fun.', 'Havana', false);
+  createMarkerForDisplay(
+      38.642805, -90.195679,
+      'I worked here during summer, 2019.', 'St Louis', false);
+
+  fetchMarkers();
+}
+
+/** 
+ * Adds a marker that shows an info window when clicked. 
+ * @param {number} lat Latitude of the marker position.
+ * @param {number} lng Longitude of the marker position.
+ * @param {string} content The content of the marker, which is displayed onclick.
```

Comment:
I would be a little specific about what exactly this "content" is

Reply:
okk